src/igvfd/types/file_set.py

```diff
@@ -376,6 +405,16 @@ def assay_titles(self, request, input_file_sets=None):
                         preferred_assay_title = measurement_set_object.get('preferred_assay_title')
                         if preferred_assay_title:
                             assay_titles.add(preferred_assay_title)
+                elif 'ConstructLibrarySet' in file_set_object.get('@type'):
+                    for sample in file_set_object.get('applied_to_samples', []):
+                        sample_object = request.embed(
+                            sample, '@@object_with_select_calculated_properties?field=file_sets')
+                        for file_set in sample_object.get('file_sets', []):
+                            file_set_object = request.embed(
+                                file_set, '@@object_with_select_calculated_properties?field=preferred_assay_title')
+                            preferred_assay_title = file_set_object.get('preferred_assay_title')
+                            if preferred_assay_title:
+                                assay_titles.add(preferred_assay_title)
```

Comment:
I was thinking if we want to calculate preferred_assay_title from input_file_sets instead? seems more straightforward to me but maybe that works equivalently.

Reply:
This calculated property, assay_titles, does calculate from the input_file_sets, in this case it is calculating from input construct library sets, which need to first go looking for the associated measurement sets.

Reply:
got it!